libs/SalesforceSDK/src/com/salesforce/androidsdk/security/Encryptor.java

```diff
@@ -120,7 +120,7 @@ public static boolean isFileSystemEncrypted() {
      * @return decrypted data
      */
     public static String decrypt(String data, String key) {
-        if (key == null || data == null) {
+        if (key == null || key.isEmpty() || data == null) {
```

Comment:
Some tests don't encrypt smartstore (they pass empty string). That used to work fine, but the external storage code blows up when trying to encrypt or decrypt files if the key is "".

I think we should support no encryption. During development, it makes sense to disable decryption to be able to introspect the database with regular sqlite viewer.